RuntimeUnityEditor.Core/Utils/Abstractions/IL2CppExtensions.cs

```diff
@@ -77,11 +78,15 @@ public static Type TryGetMonoType(this Il2CppSystem.Type il2CppType)
                                   .SelectMany(x => x.GetTypesSafe())
                                   .GroupBy(x => x.FullName))
             {
-                _cachedTypes[types.Key] = types.OrderByDescending(a =>
+                var type = types.OrderByDescending(a =>
                 {
                     var assemblyFullName = a.Assembly.FullName;
                     return assemblyFullName.StartsWith("UnityEngine.") || assemblyFullName.StartsWith("System.");
                 }).First();
+                var key = types.Key;
+                var obfuscatedNameAttribute = type.GetCustomAttribute<ObfuscatedNameAttribute>();
+                if (obfuscatedNameAttribute != null) key = obfuscatedNameAttribute.ObfuscatedName;
```

Comment:
Not sure if this can ever be a problem but just to be safe.
```suggestion
                var obfuscatedNameAttribute = type.GetCustomAttribute<ObfuscatedNameAttribute>();
                if (obfuscatedNameAttribute != null && !string.IsNullOrWhiteSpace(obfuscatedNameAttribute.ObfuscatedName)) key = obfuscatedNameAttribute.ObfuscatedName;
```